src/test/java/com/google/api/codegen/testsrc/library_gapic.yaml

```diff
@@ -227,6 +227,28 @@ interfaces:
     resource_name_treatment: STATIC_TYPES
     sample_code_init_fields:
     - series_uuid.series_string=foobar
+    code_sample_values:
+      - id: review
+        title: "Review Copy"
+        description: "Calling PublishSeries for a review copy"
+        parameters:
+          - versioning.review_copy=true
+      - id: second_edition
+        title: "Second edition"
+        description: "Calling PublishSeries for a specified edition"
+        parameters:
+          - edition=2
+    samples:
+      - molds: ["edition"]                       # TODO(vchudnov-g) use a real mold once implemented
+        values:
+          - ids: ["second_edition","review"]
+            explorer_sample: true
+            position: 1
```

Comment:
It feels like this configuration could make it difficult to reason about the complete set of explorer samples that will be presented and which order they will be in. The reader has to check for `explorer_sample` and then check for the position fields, which might or might not be there.

Reply:
YAML lists are ordered, so one alternative is to derive the position from the list position. We could also adopt the convention of listing explorer samples before non-explorer samples for readability. (It's not clear to me what the expected cardinality of samples and explorer samples is, which will have a large impact on how much of an issue this is in practice.)

Reply:
I can see how it could be, but I'm not sure it will be terribly complicated. If the user simply specified `explorer_sample` and no `position`, they samples will show in order. If they have any reason to put the values in a different order in the file than the order they want in the UI, then they need to specify `position`.

1. One alternative is to remove `explorer_sample` and only those entries with non-zero `position` are considered Explorer Samples. The drawbacks are that (a) it overloads `position` (but granted, `position` only makes sense if `explorer_sample` is specified, so they're also kinda redundant), (b) makes it less obvious what it's used for, and (c) requires the user to update all the subsequent positions in case they want to reorder (as opposed to leaving `position` unspecified and using the implicit order  in the file). 
I suppose we could address these objections by calling it `ui_position`, letting a negative value denote "use it as an ES, but toolkit can figure out the relative placement", and positive values to denote the desired relative placement.

2. A different alternative it to instead remove `position` altogether, and always use the relative order of the items in the file with `explorer_sample` set as the order in which they appear in the UI. I don't like this as much because it forces the file order to be the same as the UI order, when there's no reason they should be.

What do you think? I slightly prefer the current version (call it 0) for clarity, but I'm fine doing alternative 1 instead. I'm not convinced about alternative 2. Any other ideas?